graphql2/graphqlapp/alert.go

```diff
@@ -143,14 +146,6 @@ func (q *Query) Alerts(ctx context.Context, opts *graphql2.AlertSearchOptions) (
 			if err != nil {
 				return nil, err
 			}
-
-			// favorites only with no returned services will
-			// return an empty result set
-			if len(s.Services) == 0 {
-				return &graphql2.AlertConnection{
-					PageInfo: &graphql2.PageInfo{},
-				}, nil
-			}
```

Comment:
This was a workaround for the same reason you added `AllServices`. The consequence is if we have no favorites the response will contain alerts from all services.

Some options:
- Leave as is
- differentiate between `nil` and `len() == 0` (generally not a Go way)
- Pass a single impossible service ID (e.g. all zero UUID), probably easiest
- if `len() == 0` and `IncludeNotifiedUserID`, only include notified alerts (seems a little "magic", needs care to ensure graphql API behaves)
- add a `NotifiedUserOnly` bool to `SearchOptions` that will override the service filter. (will need to be added to cursor) maybe cleanest

I think most importantly, is the GraphQL API behavior:
- favoritesOnly = true, no favorites, and includeNotified = false should return nothing
- favoritesOnly = true, no favorites, and includeNotified = true should return only notified
- favoritesOnly = false, return everything

If we don't then you'll get all alerts, use the fav. button, and have things disappear from the alerts list which might be confusing.